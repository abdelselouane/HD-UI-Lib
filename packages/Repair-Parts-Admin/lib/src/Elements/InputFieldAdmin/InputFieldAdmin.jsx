import React from 'react';
import PropTypes from 'prop-types';
import styled from 'styled-components';

const InputFieldAdmin = (props) => {
  const { field, label, fieldError, dropDownData, toDisplay, zIndex, handleSelectedItem } = props;
  
  const DropDown = styled.div`
      position: absolute;
      border: 1px solid;
      border-color: #ccc;
      width: 100%;
      height: auto;
      background-color: #fff;
      z-index: ${props.zIndex && "1"};
    `;

  return (
    <div className={`text-input-container ${fieldError.flag && 'error'}`}>
      <input
        {...field}
      />
      { label && <label htmlFor={field.id}>{label.text}</label> }
      { dropDownData && (
        <DropDown zIndex={zIndex} >
          <table className="standard-table alternating-rows" style={{margin:'0'}}>
            <tbody>
              { dropDownData && 
                dropDownData.map( (item, index) => {
                  return (
                    <tr
                      key={index} 
                      onClick={() => handleSelectedItem(item)}
                    >
                      <td>{item[toDisplay[0]]} - {item[toDisplay[1]]}</td>  
                    </tr>); 
                })
              }               
            </tbody>
          </table>
        </DropDown>
      )}
      { fieldError.flag && <span style={{color:"#ed1c24"}}>{fieldError.msg}</span> }
    </div>
  )
}

InputFieldAdmin.propTypes = {
  field: PropTypes.object.isRequired,
  label: PropTypes.object,
  fieldError: PropTypes.object
}

export default InputFieldAdmin;