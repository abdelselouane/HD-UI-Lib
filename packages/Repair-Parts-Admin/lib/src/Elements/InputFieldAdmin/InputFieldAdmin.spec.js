import React from 'react';
import ReactDOM from 'react-dom';
import { shallow } from 'enzyme';
import InputFieldAdmin from './InputFieldAdmin';
import toJson from 'enzyme-to-json';
import * as Enzyme from 'enzyme';
import ReactSixteenAdapter from 'enzyme-adapter-react-16';

Enzyme.configure({ adapter: new ReactSixteenAdapter() });

describe('<InputFieldAdmin />', () => {
  let wrapper;
  let props;
  let item;

  afterEach(() => {
    jest.clearAllMocks();
  });

  beforeAll(() => {
    ReactDOM.createPortal = jest.fn();
    props = {
      field: {
        type: "number",
        id: "cost",
        name: "cost",
        pattern: "^\d*(\.\d{0,2})?$",
        value: '',
        required: "required",
        placeholder: "Enter cost here",
        onChange: jest.fn()
      },
      label: {
        text: "Cost"
      },
      fieldError: {
        flag: false,
        msg: ''
      },
      dropDownData: [{ brandNumber: 123, brandName: "brandName" }],
      toDisplay: ['property1', 'property2'],
      zIndex: true,
      handleSelectedItem: jest.fn()
    };
    

    // item = { brandNumber: 123, brandName: "brandName" };
    
    wrapper = shallow(<InputFieldAdmin {...props} />);
 
  
  });

  it('should render properly', () => {
    wrapper.setProps({ fieldError: { ...props.error, flag: true } });
    expect(toJson(wrapper)).toMatchSnapshot();
  });

  it('should call handleSelectedItem on row selection', () => {
    const spy = jest.spyOn(props, 'handleSelectedItem');
    const row = wrapper.find('.standard-table tbody>tr');
    row.simulate('click');
    expect(spy).toHaveBeenCalled();
  });

});