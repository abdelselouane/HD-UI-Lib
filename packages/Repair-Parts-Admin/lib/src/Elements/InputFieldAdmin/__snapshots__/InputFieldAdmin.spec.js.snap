// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<InputFieldAdmin /> should render properly 1`] = `
<div
  className="text-input-container error"
>
  <input
    id="cost"
    name="cost"
    onChange={[MockFunction]}
    pattern="^d*(.d{0,2})?$"
    placeholder="Enter cost here"
    required="required"
    type="number"
    value=""
  />
  <label
    htmlFor="cost"
  >
    Cost
  </label>
  <styled.div
    zIndex={true}
  >
    <table
      className="standard-table alternating-rows"
      style={
        Object {
          "margin": "0",
        }
      }
    >
      <tbody>
        <tr
          key="0"
          onClick={[Function]}
        >
          <td>
             - 
          </td>
        </tr>
      </tbody>
    </table>
  </styled.div>
  <span
    style={
      Object {
        "color": "#ed1c24",
      }
    }
  />
</div>
`;
